sdk/playwrighttesting/microsoft-playwright-testing/src/reporter/mptReporter.ts

```diff
@@ -313,38 +321,46 @@ class MPTReporter implements Reporter {
   private async _uploadTestResultAttachments(
     testExecutionId: string,
     testAttachments: string[],
+    otherAttachments: any[],
   ): Promise<void> {
     try {
       this.isTestRunStartSuccess = await this.promiseOnBegin;
       if (!this.isTestRunStartSuccess) {
         this._addError(`\nUnable to initialize test run report.`);
         return;
       }
-      for (const attachmentPath of testAttachments) {
-        const fileRelativePath = `${testExecutionId}/${ReporterUtils.getFileRelativePath(
-          attachmentPath,
-        )}`;
+
+      const renewSasUriIfNeeded = async (): Promise<void> => {
         if (
           this.sasUri === undefined ||
           !ReporterUtils.isTimeGreaterThanCurrentPlus10Minutes(this.sasUri)
         ) {
-          // Renew the sas uri
           this.sasUri = await this.serviceClient.createStorageUri();
           reporterLogger.info(
             `\nFetched SAS URI with validity: ${this.sasUri.expiresAt} and access: ${this.sasUri.accessLevel}.`,
           );
         }
-        await this.storageClient.uploadFile(this.sasUri.uri, attachmentPath, fileRelativePath);
+      };
+
+      for (const attachmentPath of testAttachments) {
+        const fileRelativePath = `${testExecutionId}/${ReporterUtils.getFileRelativePath(attachmentPath)}`;
+        await renewSasUriIfNeeded();
+        await this.storageClient.uploadFileAsync(this.sasUri.uri, attachmentPath, fileRelativePath);
       }
-      const rawTestResult = this.testRawResults.get(testExecutionId);
-      if (
-        this.sasUri === undefined ||
-        !ReporterUtils.isTimeGreaterThanCurrentPlus10Minutes(this.sasUri)
-      ) {
-        // Renew the sas uri
-        this.sasUri = await this.serviceClient.createStorageUri();
+
+      for (const otherAttachment of otherAttachments) {
+        await renewSasUriIfNeeded();
+        const charset = otherAttachment.contentType.match(/charset=(.*)/)?.[1];
```

Comment:
Possible array out of bounds

Reply:
![image](https://github.com/user-attachments/assets/4deb2927-516b-441f-a680-6a68692ba36b)
here we have ensured that if regex doesn't find a match in charset then we will use "utf-8", this is the same logic used by OSS also.
[OSS code]
![image](https://github.com/user-attachments/assets/01c36375-1ac1-46c8-a0e7-086345f945d3)

@puagarwa please look into this